src/core/image.js

```diff
@@ -196,6 +196,372 @@ var PDFImage = (function PDFImageClosure() {
       }
     }
   }
+
+  // Max canvas size for IOS devices
+  PDFImage.maxCanvasSize = function() {
+    return 16777216;
+  };
+
+  // Determine if we run on an IOS Mobile device
+  PDFImage.isMobile = function() {
+    if (typeof navigator !== 'undefined') {
+      return (/Mobi/.test(navigator.userAgent));
```

Comment:
> is there a way to calculate the image scale in image.js ?

yes, see #9255

Reply:
Sorry my question was not clear. What I wanted to say is how are we going to know if an image is going to render or not on a device, without trying to render it on a canvas on that device ? 
By calculating the image scale, I wanted to say, how do we automatically determine if the image needs to be scaled and by what factor.

Reply:
I can isolate the scaling functions in a new file scaling.js and call them in case it is needed from canvas.js. In order to determine if a document needs to be scaled I will try to render it, and if it does not render I will scale it by 2, then 4, 8, ... This way I would only apply the scaling in case I can't render the document. What do you think ?

Thank you,
Ioan